### PR TITLE
docs: add sphinx structure

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
docs: add sphinx structure

Add base sphinx structure to serve our documents as html pages;
The structure is similar to what cloud-init does on their side, but there are still some points that we need to discuss:

1. What should we do to the github specific notation on the documents ? Like notes and warnings
2. Do you want to use a different structure ?
3. Should our developer documentation be included ?

## Test Steps
Run `tox -e docs` and verify that the html pages were generated on `docs/build/html`
Please verify that the content is working as expected. specially the redirects in the pages

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by CalvoM** [comment]
I just noticed the tox command is `tox -e doc` and  not `tox -e docs` .

**Comment by CalvoM** [comment]
Could `docs/build` be included in the `make clean` command?

File: tox.ini [hunk 1]
```diff
@@ -33,6 +33,7 @@ deps =
     isort: -rdev-requirements.txt
     behave: -rintegration-requirements.txt
     shellcheck: -rdev-requirements.txt
+    doc: -rdoc-requirements.txt
```
Comment (renanrodrigo) [inline tox.ini:36]: Should either be `docs` here, or `doc` in the command section below

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "ua-docs"
+copyright = "2022, Lucas Moura"
```
Comment (renanrodrigo) [inline docs/source/conf.py:21]: Shouldn't the copyright be `Canonical`? (I'm guessing, not sure)

**Review by renanrodrigo — APPROVED** [review]
This iteration looks great (:
Just a couple inline comments to address

File: docs/source/index.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- `Common Criteria EAL2 Certification Tooling <https://ubuntu.com/security/cc>`_
+- `CIS Benchmark Audit Tooling <https://ubuntu.com/security/cis>`_
+- `Ubuntu Security Guide (USG) Tooling <https://ubuntu.com/security/certifications/docs/usg>`_
+- `Ubuntu Extended Security Maintenance (ESM) <https://ubuntu.com/security/esm>`_
+- `Robot Operating System (ROS) Extended Security Maintenance <https://ubuntu.com/robotics/ros-esm>`_
+- `FIPS 140-2 Certified Modules (and optional non-certified patches <(https://ubuntu.com/security/fips>`_
+- `Livepatch <https://ubuntu.com/security/livepatch>`_
+
+If you need any of those services for your machine, UA is the right tool for you.
+Furthermore, UA is already installed on every Ubuntu system. Try it out by running `ua help`!
```
Comment (orndorffgrant) [inline docs/source/index.rst:24]: `ua help` gets rendered as _italics_. I think in RST, inline code is

```
``ua help``
```

File: docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
+.. ua-docs documentation master file, created by
+   sphinx-quickstart on Wed Jun 15 14:45:09 2022.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
```
Comment (orndorffgrant) [inline docs/source/index.rst:5]: we can probably remove this

**Review by orndorffgrant — APPROVED** [review]
Looks good! Just some minor things

File: docs/source/conf.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "ua-docs"
```
Comment (orndorffgrant) [inline docs/source/conf.py:20]: Should this be "Ubuntu Advantage Client" as the name of the project?

**Comment by lucasmoura** [comment]
@orndorffgrant updated

**Comment by orndorffgrant** [comment]
Yep sorry, my approval is still valid. Thanks for the fixes!

**Pull request merged**